nvflare/app_common/executors/multi_process_executor.py

```diff
@@ -158,116 +175,103 @@ def _initialize_multi_process(self, fl_ctx: FLContext):
-                            responses[index] = True
-                    index += 1
-                if received_all:
-                    break
-        except:
-            self.log_exception(fl_ctx, "error initializing multi_process executor")
-
-    def _relay_fire_event(self, listen_port, fl_ctx: FLContext):
-        address = ("localhost", int(listen_port))
-        listener = Listener(address, authkey=CommunicationMetaData.PARENT_PASSWORD.encode())
-        conn = listener.accept()
-
-        while not self.stop_execute:
-            try:
-                if conn.poll(0.1):
-                    data = conn.recv()
-                    event_type = data[CommunicationMetaData.EVENT_TYPE]
-                    rank_number = data[CommunicationMetaData.RANK_NUMBER]
-
-                    with self.relay_lock:
-                        fl_ctx.props.update(data[CommunicationMetaData.FL_CTX].props)
-
-                        fl_ctx.set_prop(FLContextKey.FROM_RANK_NUMBER, rank_number, private=True, sticky=False)
-                        fl_ctx.set_prop(
-                            FLContextKey.EVENT_ORIGIN_SITE,
-                            CommunicateData.SUB_WORKER_PROCESS,
-                            private=True,
-                            sticky=False,
-                        )
-                        engine = fl_ctx.get_engine()
-                        engine.fire_event(event_type, fl_ctx)
-
-                        return_data = {CommunicationMetaData.FL_CTX: get_serializable_data(fl_ctx)}
-                    conn.send(return_data)
-            except:
-                self.logger.warning("Failed to relay the fired events from rank_processes.")
-
-    def _create_connection(self, open_port):
-        conn = None
-        while not conn:
-            try:
-                address = ("localhost", open_port)
-                conn = Client(address, authkey=CommunicationMetaData.CHILD_PASSWORD.encode())
-            except BaseException:
-                time.sleep(1.0)
-                pass
-        return conn
+                self.engine.fire_event(event_type, fl_ctx)
+                return_data = {CommunicationMetaData.FL_CTX: get_serializable_data(fl_ctx)}
+                return F3make_reply(ReturnCode.OK, "", return_data)
 
     def execute(self, task_name: str, shareable: Shareable, fl_ctx: FLContext, abort_signal: Signal) -> Shareable:
         if not self.executor:
             raise RuntimeError("There's no executor for task {}".format(task_name))
 
-        return self._execute_multi_process(
-            task_name=task_name, shareable=shareable, fl_ctx=fl_ctx, abort_signal=abort_signal
-        )
+        self.execute_complete = False
+
+        self._execute_multi_process(task_name=task_name, shareable=shareable, fl_ctx=fl_ctx, abort_signal=abort_signal)
+
+        while not self.execute_complete:
+            time.sleep(0.2)
+        return self.execute_result
 
     def _execute_multi_process(
```

Comment:
This method no longer needs to return a Shareable.
Should we remove the return statements?

Reply:
Yeah, the cellnet way implementation does not require the` _execute_multi_process() `to return a Shareable any more. It gets the result in a different way. but the return does not hurt any way.